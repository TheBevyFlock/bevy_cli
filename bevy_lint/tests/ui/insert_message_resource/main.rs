#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::insert_message_resource)]

use bevy::prelude::*;

#[derive(Message)]
struct Foo;

fn main() {
    let mut app = App::new();
    App::new().init_resource::<Messages<Foo>>();
    //~^ ERROR: called `App::init_resource::<Messages<Foo>>()` instead of
    // `App::add_message::<Foo>()`

    App::init_resource::<Messages<Foo>>(&mut app);
    //~^ ERROR: called `App::init_resource::<Messages<Foo>>(&mut app)` instead of
    // `App::add_message::<Foo>(&mut app)`

    App::new().insert_resource::<Messages<Foo>>(Default::default());
    //~^ ERROR: called `App::insert_resource::<Messages<Foo>>(Default::default())` instead of
    // `App::add_message::<Foo>()`

    App::insert_resource::<Messages<Foo>>(&mut app, Default::default());
    //~^ ERROR: called `App::insert_resource::<Messages<Foo>>(&mut app, Default::default())` instead
    // of `App::add_message::<Foo>(&mut app)`

    // Make sure the correct type is detected, even when not explicitly passed to
    // `insert_resource()`.
    let implied_event: Messages<Foo> = Default::default();
    App::new().insert_resource(implied_event);
    //~^ ERROR: called `App::insert_resource(implied_event)` instead of `App::add_message::<Foo>()`

    // Ensure the lint can be muted by annotating the expression.
    #[allow(bevy::insert_message_resource)]
    App::new().init_resource::<Messages<Foo>>();
}
