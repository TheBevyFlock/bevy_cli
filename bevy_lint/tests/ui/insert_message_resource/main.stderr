error: called `App::init_resource::<Messages<Foo>>()` instead of `App::add_message::<Foo>()`
  --> tests/ui/insert_message_resource/main.rs:12:16
   |
12 |     App::new().init_resource::<Messages<Foo>>();
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/insert_message_resource/main.rs:3:9
   |
 3 | #![deny(bevy::insert_message_resource)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: inserting an `Messages` resource does not fully setup that message
   |
12 -     App::new().init_resource::<Messages<Foo>>();
12 +     App::new().add_message::<Foo>();
   |

error: called `App::init_resource::<Messages<Foo>>(&mut app)` instead of `App::add_message::<Foo>(&mut app)`
  --> tests/ui/insert_message_resource/main.rs:16:5
   |
16 |     App::init_resource::<Messages<Foo>>(&mut app);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Messages` resource does not fully setup that message
   |
16 -     App::init_resource::<Messages<Foo>>(&mut app);
16 +     App::add_message::<Foo>(&mut app);
   |

error: called `App::insert_resource::<Messages<Foo>>(Default::default())` instead of `App::add_message::<Foo>()`
  --> tests/ui/insert_message_resource/main.rs:20:16
   |
20 |     App::new().insert_resource::<Messages<Foo>>(Default::default());
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Messages` resource does not fully setup that message
   |
20 -     App::new().insert_resource::<Messages<Foo>>(Default::default());
20 +     App::new().add_message::<Foo>();
   |

error: called `App::insert_resource::<Messages<Foo>>(&mut app, Default::default())` instead of `App::add_message::<Foo>(&mut app)`
  --> tests/ui/insert_message_resource/main.rs:24:5
   |
24 |     App::insert_resource::<Messages<Foo>>(&mut app, Default::default());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Messages` resource does not fully setup that message
   |
24 -     App::insert_resource::<Messages<Foo>>(&mut app, Default::default());
24 +     App::add_message::<Foo>(&mut app);
   |

error: called `App::insert_resource(implied_event)` instead of `App::add_message::<Foo>()`
  --> tests/ui/insert_message_resource/main.rs:31:16
   |
31 |     App::new().insert_resource(implied_event);
   |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: inserting an `Messages` resource does not fully setup that message
   |
31 -     App::new().insert_resource(implied_event);
31 +     App::new().add_message::<Foo>();
   |

error: aborting due to 5 previous errors

